### PR TITLE
PHP 8.0 Upgrade

### PR DESCRIPTION
**Pull request by aidan-casey**
Not entirely sure what the appetite for change is here, but thought I would start a pull request to see if it was welcomed.

This PR does the following:

- Enforces parameter types through modern PHP type hinting thus removing the need for some assertion classes.
- Enforces property types through modern PHP type hinting.
- Removes unnecessary doc blocks.
- Updates directory structure to match Laravel recommendations.
- Adds tests for 100% code coverage.
- Requires PHP 8.0.
- Updates test workflow to run in about 13 seconds as opposed to 6 minutes.

### CODE REVIEW

**Comment by denisdulici** [comment]
I follow [Laravel](https://laravel.com/docs/9.x/releases#support-policy) for the PHP version requirement, so can't merge this PR as is.

**Comment by aidan-casey** [comment]
@d

**Comment by aidan-casey** [comment]
@denisdulici if I make this PHP 8.0 compatible would you be of the mind to merge?

**Comment by denisdulici** [comment]
Sure, why not.

**Comment by aidan-casey** [comment]
@denisdulici - I have reverted the PHP 8.1 specific changes so that this will support PHP 8 and 8.1. Note that I have also updated the directory structure to match Laravel's recommendations and update the testing workflow to gain some faster run times.

**Comment by denisdulici** [comment]
Thanks for your effort. Can you please keep the `Laravel 8` compatibility? At least until Laravel [drops](https://laravel.com/docs/releases#support-policy) the bug fixes.

**Comment by denisdulici** [comment]
No, they're not the same thing, and this package requires at least `PHP 8` + `Laravel 8` since the 2.0 version.

**Comment by aidan-casey** [comment]
@denisdulici - Sorry I misunderstood based on your comment earlier:

> I follow [Laravel](https://laravel.com/docs/9.x/releases#support-policy) for the PHP version requirement, so can't merge this PR as is.

I have updated the dependencies as requested and also added them to the testing workflow to ensure everything is in order.

File: tests/TestCase.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Akaunting\Tests\Money;
```
Comment (denisdulici) [inline tests/TestCase.php:3]: The namespace must be `Akaunting\Money\Tests`. Same issue for all test classes.

**Review by denisdulici — CHANGES_REQUESTED** [review]

File: src/Provider.php [hunk 2]
```diff
@@ -8,15 +8,10 @@
...
-     * Bootstrap the application services.
-     *
-     * @return void
-     */
-    public function boot()
+    public function boot(): void
     {
         $this->publishes([
-            __DIR__ . '/Config/money.php' => config_path('money.php'),
+            __DIR__ . '/../money.php' => config_path('money.php'),
```
[the remark this answers is not in the document]
Comment (aidan-casey) [reply src/Provider.php]: My bad, good catch! This is resolved in the latest commit.

File: tests/TestCase.php (hunk 1 above)
Comment (aidan-casey) [reply tests/TestCase.php]: Fixed.

**Comment by denisdulici** [comment]
Thanks @aidan-casey for this huge contribution 🙏

**Comment by aidan-casey** [comment]
No problem. Thanks for the feedback throughout!

**Pull request merged**